x-pack/heartbeat/monitors/browser/source/local.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package source
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
+
+	"github.com/otiai10/copy"
+)
+
+type LocalSource struct {
+	OrigPath    string `config:"path"`
+	workingPath string
+	BaseSource
+}
+
+func (l *LocalSource) Fetch() (err error) {
+	if l.workingPath != "" {
+		return nil
+	}
+	l.workingPath, err = ioutil.TempDir("/tmp", "elastic-synthetics-")
+	if err != nil {
+		return fmt.Errorf("could not create tmp dir: %w", err)
+	}
+
+	err = copy.Copy(l.OrigPath, l.workingPath)
+	if err != nil {
+		return fmt.Errorf("could not copy suite: %w", err)
+	}
+
+	dir, err := getSuiteDir(l.workingPath)
```

Comment:
This section dealing with deps was cut / pasted from synthexec.go, it makes much more sense here.

We'll probably share it in the future when we have github / zip sources, but let's not refactor prematurely